pkg/deploy/deploy.go

```diff
@@ -488,4 +388,137 @@ func OperatorExists(cli client.Client, operatorPrefix string) (bool, error) {
 	return false, nil
 }
 
+func getResource(ctx context.Context, cli client.Client, obj *unstructured.Unstructured) (*unstructured.Unstructured, error) {
+	found := &unstructured.Unstructured{}
+	// Setting gvk is required to do Get request
+	found.SetGroupVersionKind(obj.GroupVersionKind())
+	err := cli.Get(ctx, types.NamespacedName{Name: obj.GetName(), Namespace: obj.GetNamespace()}, found)
+	if apierrs.IsNotFound(err) {
+		return nil, nil
+	}
+	return found, err
```

Comment:
If it's not NotFound it will return the error and blank object. Hope, it's ok for you. It does not break the code (the check in manageResource for err will filter it).